DoSomethingArchive/legacy-website
# Campaign Status field updates

#### What's this PR do?
- Disables the campaign status field on campaign nodes so editors can not use it.
- Makes the campaign status field on campaign nodes translatable.
- Adds a presave hook to campaign runs that will update campaign status when a user saves a campaign run node.
- Creates a helper function that takes a start and end date and determines if a campaign is active or not
- Slightly outside of the scope of this issue, but I was getting some preprocess errors on pages without subtitle and partners fields that I fixed up.
- Creates a cron job that updates the campaign status field based on run dates.
#### Where should the reviewer start?

The main business is in these here files, there rest is feature export stuff.

`dosomething_campaign_run.module`
`dosomething_helpers.module`
`dosomething_campaign_run.cron.inc`
#### How should this be manually tested?

Create or edit a campaign run, and run dates, hit save. Go to the campaign you tied the run to, the campaign status field should be disabled, but you should be able to see the current status of the campaign per translation. 

Status would be `active` when the run has a start date and no end date
Status would be `closed` when the run end date is before `now`

The cron job is a bit trickier to test, it really only comes in handy when there is a campaign run dates set in the future or past. The cron job should update the status of the campaign node to closed or active based on the dates without a use having to manually save the campaign run again.
#### Any background context you want to provide?

Campaign runs...amirite?
#### What are the relevant tickets?

Fixes #5989
